packages/seedless-onboarding-controller/src/SeedlessOnboardingController.ts

```diff
@@ -408,11 +465,23 @@ export class SeedlessOnboardingController extends BaseController<
     }
   }
 
+  /**
+   * Verify the password validity by decrypting the vault.
+   *
+   * @param password - The password to verify.
+   * @throws {Error} If the password is invalid or the vault is not initialized.
+   */
+  async #verifyPassword(password: string) {
+    if (!this.state.vault) {
+      throw new Error(SeedlessOnboardingControllerError.VaultError);
+    }
+    await this.#vaultEncryptor.decrypt(password, this.state.vault);
+  }
+
   /**
    * Unlocks the encrypted vault using the provided password and returns the decrypted vault data.
    * This method ensures thread-safety by using a mutex lock when accessing the vault.
```

Comment:
Function description needs to be updated since we don't provide password anymore as arg